src/main/java/org/cbioportal/web/PublicVirtualStudiesController.java

```diff
@@ -0,0 +1,209 @@
+public class PublicVirtualStudiesController {
+
+    private static final Logger LOG = LoggerFactory.getLogger(PublicVirtualStudiesController.class);
+
+    public static final String ALL_USERS = "*";
+
+    private final String requiredPublisherApiKey;
+
+    private final SessionServiceRequestHandler sessionServiceRequestHandler;
+
+    private final String sessionServiceURL;
+
+    private final CancerTypeService cancerTypeService;
+
+    private final VirtualStudyPermissionService virtualStudyPermissionService;
+
+    public PublicVirtualStudiesController(
+        @Value("${session.endpoint.publisher-api-key:}") String requiredPublisherApiKey,
+        SessionServiceRequestHandler sessionServiceRequestHandler,
+        @Value("${session.service.url:}") String sessionServiceURL,
+        CancerTypeService cancerTypeService,
+        VirtualStudyPermissionService virtualStudyPermissionService
+    ) {
+        this.requiredPublisherApiKey = requiredPublisherApiKey;
+        this.sessionServiceRequestHandler = sessionServiceRequestHandler;
+        this.sessionServiceURL = sessionServiceURL;
+        this.cancerTypeService = cancerTypeService;
+        this.virtualStudyPermissionService = virtualStudyPermissionService;
+    }
+
+    @GetMapping
+    @ApiResponse(responseCode = "200", description = "OK", content = @Content(schema = @Schema(implementation = VirtualStudy.class)))
+    public ResponseEntity<List<VirtualStudy>> getPublicVirtualStudies() {
+        //TODO move this logic to sessionServiceRequestHandler?
+        BasicDBObject basicDBObject = new BasicDBObject();
+        basicDBObject.put("data.users", ALL_USERS);
+        ResponseEntity<List<VirtualStudy>> responseEntity = new RestTemplate().exchange(
+            sessionServiceURL + "/virtual_study/query/fetch",
+            HttpMethod.POST,
+            new HttpEntity<>(basicDBObject.toString(), sessionServiceRequestHandler.getHttpHeaders()),
+            new ParameterizedTypeReference<>() {
+            });
+
+        List<VirtualStudy> virtualStudies = responseEntity.getBody();
+        virtualStudyPermissionService.filterOutForbiddenStudies(virtualStudies);
+        return new ResponseEntity<>(virtualStudies, HttpStatus.OK);
+    }
+
+    @PostMapping
+    @ApiResponse(responseCode = "200", description = "OK", content = @Content(schema = @Schema(implementation = VirtualStudy.class)))
+    public ResponseEntity<VirtualStudy> publishVirtualStudyData(
+        @RequestBody VirtualStudyData virtualStudyData,
+        @RequestHeader(value = "X-PUBLISHER-API-KEY") String providedPublisherApiKey,
+        @RequestParam(required = false) String typeOfCancerId,
+        @RequestParam(required = false) String pmid
+    ) {
+        if (requiredPublisherApiKey == null
+            || requiredPublisherApiKey.isBlank()
+            || !requiredPublisherApiKey.equals(providedPublisherApiKey)) {
+            return new ResponseEntity<>(null, HttpStatus.UNAUTHORIZED);
+        }
```

Comment:
How we usually handle these is to utilize the GlobalExceptionHandler class 

That captures an exception thrown and sends a response back. Example:
```  @ExceptionHandler(DataAccessTokenProhibitedUserException.class)
    public ResponseEntity<ErrorResponse> handleDataAccessTokenProhibitedUserException() {
        ErrorResponse response = new ErrorResponse("You are prohibited from using Data Access Tokens");
        return new ResponseEntity<>(response, HttpStatus.UNAUTHORIZED);
    }

```

Reply:
Same thing for the others that are returning a null body

Reply:
@haynescd Are you saying that it's not necessary to have the `requiredPublisherApiKey == null` clause in the condition?

Reply:
I can't find a handler for NPE in `GlobalExceptionHandler`. Shall we add one?